pyaerocom/colocation_auto.py

```diff
@@ -383,28 +385,32 @@ def run(self, var_name=None, **opts):
         finally:
             self._close_log()
 
-    def get_gridded_reader(self):
-        """
-        Returns the class of the reader used for model data.
+    def instantiate_gridded_reader(self, what):
         """
-        try:
-            reader = self.SUPPORTED_GRIDDED_READERS[self.gridded_reader_id]
-        except KeyError as e:
-            raise NotImplementedError('Reader {} is not supported: {}'.format(self.gridded_reader_id, e))
-        return reader
-
+        Create reader for model or observational gridded data.
 
-    def instantiate_model_reader(self):
-        """
-        Create reader object based on reader id.
+        Parameters
+        ----------
+        what : str
+            Type of reader. ("model" or "obs")
         """
-        reader_class = self.get_gridded_reader()
-        reader = reader_class(data_id=self.model_id)
+        if what == 'model':
+            id = self.model_id
```

Comment:
Consider renaming `id` to `data_id` as the former is a python builtin method.

Reply:
Will do!